bgzf/reader.go

```diff
@@ -4,22 +4,84 @@ import (
 	"bufio"
 	"bytes"
 	"compress/gzip"
+	"io"
 	"log"
 	"math"
 	"os"
 )
 
-// Reader is the decompressor for BGZF files which is at it's core
+// BlockReader is the decompressor for BGZF files which is at its core
 // a gzip reader. The BGZF reader retains a *os.File to use for
 // seek functionality as well as a *bufio.Reader to implement the
 // io.ByteReader interface, required for reading gzip files as
 // independent concatenated blocks, rather than a multistream.
-type Reader struct {
+type BlockReader struct {
 	file         *os.File
 	buf          *bufio.Reader
 	decompressor *gzip.Reader
 }
 
+// Read wraps a BlockReader and provides the Next and Read functions
+// that automatically read a new block when the previous block has
+// been completely read.
+type Reader struct {
```

Comment:
I don't see "Reader" used right now.  I see bams using the blockReader, but I am guessing that you have a vision for using this somewhere else